scylla-macros/src/deserialize/value.rs

```diff
@@ -83,26 +90,195 @@ pub(crate) fn deserialize_value_derive(
     tokens_input: TokenStream,
 ) -> Result<syn::ItemImpl, syn::Error> {
     let input = syn::parse(tokens_input)?;
+    let input: syn::DeriveInput = input;
+
+    match &input.data {
+        syn::Data::Enum(data_enum) => deserialize_value_derive_enum(&input, data_enum),
+        syn::Data::Struct(_) => {
+            let implemented_trait: syn::Path = parse_quote!(DeserializeValue);
+            let implemented_trait_name = implemented_trait
+                .segments
+                .last()
+                .unwrap()
+                .ident
+                .unraw()
+                .to_string();
```

Comment:
:wrench: Justify the `unwrap()`. Why can we be sure that there is at least one segment?